### PR TITLE
Merging Task/amazon-book with main

### PR DESCRIPTION
**Pull request by yedigaryansem**

### CODE REVIEW

File: src/main/java/pages/amazon/CategoryByBooksPage.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public CategoryByBooksPage(WebDriver driver){
+        this.driver = driver;
+    }
+
+    public WebElement waitForElement(By loc){
+        WebDriverWait wait = new WebDriverWait(this.driver,10);
+
+        return wait.until(ExpectedConditions.visibilityOfElementLocated(loc));
+    }
```
Comment (David91) [inline src/main/java/pages/amazon/CategoryByBooksPage.java]: 1. This method shouldn't be public. It is not allowed ( it is against POM principles ) to pass a locator from outside of the page, as the locator/webelement should be inside the page.
2. Why to have such kind of method ? What is gives to you ?

File: src/main/java/pages/amazon/CategoryByBooksPage.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    public List<WebElement> waitForElementsByCount(By loc, int count){
+        WebDriverWait wait = new WebDriverWait(this.driver,10);
+
+        return wait.until(ExpectedConditions.numberOfElementsToBeMoreThan(loc,count-1));
+    }
```
Comment (David91) [inline src/main/java/pages/amazon/CategoryByBooksPage.java]: Same comment as previous one.

File: src/main/java/pages/amazon/CategoryByBooksPage.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    public List<String> getAuthorsNameFromBooks(int countOFBooks){
+        List<WebElement> namesElem = waitForElementsByCount(authorsNameDir,countOFBooks);
```
Comment (David91) [inline src/main/java/pages/amazon/CategoryByBooksPage.java:43]: Why not write just 
List<WebElement> namesElem = wait.until(ExpectedConditions.numberOfElementsToBeMoreThan(authorsNameDir,countOFBooks-1)); ?

File: src/main/java/pages/amazon/CategoryByKindleStorePage.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        return wait.until(ExpectedConditions.numberOfElementsToBeMoreThan(loc, count-1));
+    }
+
+    public String getBooksByAuthorText(){
+        return waitForElement(booksByAuthorText).getText();
+    }
+    public void changeBooksSortingFromLowToHigh(){
+        waitForElement(sortingDropDownButton).click();
+        waitForElement(priceLowToHighSDDButton).click();
+    }
```
Comment (David91) [inline src/main/java/pages/amazon/CategoryByKindleStorePage.java:53]: This is global comment. Why you wait for everything in pages ?
Once you've made sure that page is loaded, you can access to all elements in page.

File: src/main/java/pages/amazon/HomePage.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class HomePage {
+    private final WebDriver driver;
+
+    private final String pageLink = "https://www.amazon.com/";
```
Comment (David91) [inline src/main/java/pages/amazon/HomePage.java]: It'd be better for each page keep it's part of URL. URL just an environment, against which tests will run, but each page has it's relative path regardless environment. So if amazon's home page URL is "BASE_URL + "/home", then you can keep inside HomePage only "/home" part of URL.

File: src/test/java/amazonTestSuite/amazonFeaturesTest.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        Assert.assertEquals(actualText,expectedText,String.format("The actual name does not match with the expected name %s",testName));
+
+        categoryByKindleStorePage.changeBooksSortingFromLowToHigh();
+        List<Float> prices = categoryByKindleStorePage.getAllBooksPrice(12);
+        for (int i = 0; i < prices.size()-1; ++i) {
+            Assert.assertTrue(prices.get(i)<=prices.get(i+1),"The sorting drop down is incorrect");
+        }
+    }
+
+    @Test(enabled = true)
```
Comment (David91) [inline src/test/java/amazonTestSuite/amazonFeaturesTest.java]: Test is enabled by default, so no need to explicitly specify it.

**Review by David91 — CHANGES_REQUESTED** [review]

File: src/test/java/amazonTestSuite/amazonFeaturesTest.java [hunk 7]
```diff
@@ -44,6 +44,7 @@ public void authorsNameTest(){
         homePage.searchBySearchingBarByText(testName);
 
         CategoryByBooksPage categoryByBooksPage = new CategoryByBooksPage(driver);
+        categoryByBooksPage.getPage("s?k=Tsugumi+Ohba&i=stripbooks-intl-ship&ref=nb_sb_noss");
```
Comment (David91) [inline src/test/java/amazonTestSuite/amazonFeaturesTest.java:47]: Remind me talk about this later. This shouldn't be so, but don't fix this for now.

File: src/test/java/amazonTestSuite/amazonFeaturesTest.java [hunk 8]
```diff
@@ -53,6 +54,7 @@ public void authorsNameTest(){
         categoryByBooksPage.clickOnAuthorsNamedFromBook();
 
         CategoryByKindleStorePage categoryByKindleStorePage = new CategoryByKindleStorePage(driver);
+        categoryByKindleStorePage.getPage("Tsugumi-Ohba/e/B0035POVWA?ref=sr_ntt_srch_lnk_1&qid=1617348759&sr=1-1");
```
Comment (David91) [inline src/test/java/amazonTestSuite/amazonFeaturesTest.java:57]: The same comment for here.

**Review by David91 — APPROVED** [review]

File: src/test/java/amazonTestSuite/amazonFeaturesTest.java (hunk 8 above)
Comment (yedigaryansem) [reply src/test/java/amazonTestSuite/amazonFeaturesTest.java:57]: Ok

File: src/test/java/amazonTestSuite/amazonFeaturesTest.java (hunk 7 above)
Comment (yedigaryansem) [reply src/test/java/amazonTestSuite/amazonFeaturesTest.java:47]: Ok

**Pull request merged**